test/testExport.js

```diff
@@ -1,13 +1,216 @@
 "use strict";
 
-const analyticsExport = require("../analytics_export/analyticsExport.js");
+const proxyquire = require("proxyquire");
+const { assert } = require("chai");
+const { beforeEach, describe, it } = require("mocha");
+const { spy, stub } = require("sinon");
 
-const chai = require("chai");
+const { AnalyticsExport } = require("../analytics_export/analyticsExport");
+const { RequestError } = require("../analytics_export/requestError");
 
-chai.should();
+describe("Analytics export", () => {
+  describe("getAllowedDimensionsPerMeasure", () => {
+    let mockItcClient;
+    let analyticsExport;
 
-describe("Export Functions", function () {
-    it("should pass test", function () {
-        analyticsExport.startExport().should.equal("test");
+    beforeEach(() => {
+      mockItcClient = {
+        getSettings: () => {},
+      };
+      analyticsExport = new AnalyticsExport(
+        mockItcClient,
+        "project",
+        "dataset",
+        "123",
+        "Firefox",
+      );
     });
+
+    it("should fail if given date with no data", async () => {
+      stub(mockItcClient, "getSettings").callsArgWith(0, null, {
+        configuration: {
+          dataStartDate: "2020-01-02T00:00:00",
+          dataEndDate: "2020-01-03T00:00:00",
+        },
+      });
+      analyticsExport.client = mockItcClient;
+
+      const getAllowedDimensionsSpy = spy(
+        analyticsExport,
+        "getAllowedDimensionsPerMeasure",
+      );
+
+      let throwCount = 0;
+
+      try {
+        await getAllowedDimensionsSpy(
+          Date.parse("2020-01-01"),
+          Date.parse("2020-01-02"),
+        );
+      } catch (e) {
+        throwCount += 1;
+      }
```

Comment:
I'm not entirely clear on why you need three test cases here. Also shouldn't we use jest's built in test for throwing an error? https://jestjs.io/docs/en/expect#tothrowerror

Reply:
I think I was trying to test before, after, and completely out of the boundaries but I mixed up the dates.  And yes should use the builtin things for chai

Reply:
I also noticed that the wrong error is being thrown in the run so that needs to be fixed and checked